### ISSUE TITLE
The dataset are inconsistent

### ISSUE THREAD

**Issue by 110wuqu** [simonalexanderson/StyleGestures#41]
I downloaded the https://kth-my.sharepoint.com/:f:/g/personal/simonal_ug_kth_se/EuzCqSiScf5EvwWgZJ8EZ3wB24oPsKca9klzGzxzV-4N6g?e=BG1ELf and running the training code found that the dimensions of the test set and training set are inconsistent  .
input_data: (13710, 80, 66)
test_data: (31, 100, 66)

**Comment by ghenter** [comment]
My (potentially incorrect) impression of this is that this need not be a big deal. The data has 20 frames (poses) per second. Assuming that the second dimension is time, this means that models are trained on 4-second sequences (80 frames), and tested on 5-second sequences (100 frames). MoGlow is an autoregressive model and is not restricted to data with a particular, fixed length, so it has a well defined behaviour both for 4 and 5-second sequences.

Does this make sense, or am I perhaps missing something here?

**Comment by ghenter** [comment]
I don't have detailed knowledge about the code unfortunately, but I don't recognise where the number "694" comes from, nor how it is related to the numbers that were posted earlier.
> input_data: (13710, 80, 66)
> test_data: (31, 100, 66)

**Comment by 110wuqu** [comment]
In addition, I would like to ask whether it is normal for likehood_loss to be negative all the time during training

**Comment by ghenter** [comment]
>I would like to ask whether it is normal for likehood_loss to be negative all the time during training

The loss used to train our systems is the logarithm of the likelihood, a.k.a. the negative log-likelihood, or NLL. The NLL for a continuous-valued distribution, as considered here, can in principle attain any value on the real line. In particular, negative loss values are normal and expected in our training, and we see them all the time. Please refer to the training curves in [our paper at INNF+ 2020](https://invertibleworkshop.github.io/INNF_2020/accepted_papers/pdfs/25.pdf) to see what example training curves can look like.

**Comment by ghenter** [comment]
>The error appears here
h = self.f(z1_cond.permute(0, 2, 1)).permute(0, 2, 1)
z1_cond.shape is ( ,694, 70)
z1_cond = torch.cat((z1, cond), dim=1)
z1.shape is (,63//2,70)
cond.shape is (,663,70)

I'm sorry, but I cannot easily see the origin of the issue in this snippet that you shared. Since I did not write this code and am not a developer, I don't think I can debug the issue you are having effectively via GitHub comments. @simonalexanderson, might know better, but I think he is busy with other work right now.

**Comment by ghenter** [comment]
@110wuqu Based on the training curves that you posted in issue #43 a little while ago, I am getting the impression that you have managed to solve and/or work around the problems that you asked about in this issue. As a courtesy, could you please:

1. Write what the problem was and how you solved it, so that other people who come here with the same issue will find the solution here.
2. Close the issue, if appropriate.

**Comment by 110wuqu** [comment]
Sorry, because it was too long ago, I have forgotten how I solved this problem in the beginning, there is no problem in the code of the project itself, it should be my careless problem.
